Unable to iterage through a returned channel

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
)

func someChan() <-chan struct{} {
	c := make(chan struct{}, 1)
	c <- struct{}{}
	return c
}

func main() {
	for _ = range someChan() {
		fmt.Println("success")
		return
	}
}
```

Expected result:
```console
$ go run ./sample.go
success
```

Got:
```console
$ yaegi ./sample.go
cfg.go:129: case8/main.go:14:27: in call
scope.go:154: nil type
nil type
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0000e2000, 0x40)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc000219df0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:328 +0xbc
panic(0xc99360, 0xc0000a47e0)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
log.Panic(0xc000219428, 0x1, 0x1)
	/usr/lib/go/src/log/log.go:351 +0xac
github.com/containous/yaegi/interp.(*scope).add(0xc0000d0460, 0x0, 0x1)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/scope.go:154 +0x1bf
github.com/containous/yaegi/interp.(*Interpreter).cfg.func1(0xc0000ceb00, 0xc000219c98)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/cfg.go:130 +0x178b
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ceb00, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:187 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ce700, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:191 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ce600, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:191 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ce500, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:191 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ce000, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:191 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000be100, 0xc000219c98, 0xc000219c68)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:191 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc000370000, 0xc0000be100, 0x7fff712ad1a8, 0xd, 0xc0000b2088, 0x4, 0x7fff712ad1a8, 0xd, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/cfg.go:55 +0x160
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000370000, 0xc0000c6000, 0xd0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:355 +0x28d
main.main()
	/home/jlevesy/code/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:143 +0x553
```

--- a/_test/range7.go
+++ b/_test/range7.go
@@ -1,0 +1,21 @@
+package main
+
+import (
+	"fmt"
+)
+
+func someChan() <-chan struct{} {
+	c := make(chan struct{}, 1)
+	c <- struct{}{}
+	return c
+}
+
+func main() {
+	for _ = range someChan() {
+		fmt.Println("success")
+		return
+	}
+}
+
+// Output:
+// success

--- a/interp/scope.go
+++ b/interp/scope.go
@@ -135,6 +135,9 @@ func (s *scope) rangeChanType(n *node) *itype {
 			return t
 		}
 	}
+	if c := n.child[1]; c.typ != nil && c.typ.cat == chanT {
+		return n.child[1].typ
+	}
 	return nil
 }
 

--- a/interp/scope.go
+++ b/interp/scope.go
@@ -136,7 +136,7 @@ func (s *scope) rangeChanType(n *node) *itype {
 		}
 	}
 	if c := n.child[1]; c.typ != nil && c.typ.cat == chanT {
-		return n.child[1].typ
+		return c.typ
 	}
 	return nil
 }

